plugin/imports.go

```diff
@@ -33,58 +34,89 @@ func CleanImports(pFileText *string) *string {
+// unless the package is already imported for this file. Either way, it returns
+// the package alias
+func (p *OrmPlugin) NewImport(packagePath string) string {
+	subpath := packagePath[strings.LastIndex(packagePath, "/")+1:]
+	// package will always be suffixed with an integer to prevent any collisions
+	// with standard package imports
+	for i := 1; ; i++ {
+		newAlias := fmt.Sprintf("%s%d", strings.Replace(subpath, ".", "_", -1), i)
+		if pkg, ok := p.GetFileImports().packages[newAlias]; ok {
+			if packagePath == pkg.packagePath {
+				return pkg.alias
+			}
+		} else {
+			p.GetFileImports().packages[newAlias] = &pkgImport{packagePath: packagePath, alias: newAlias}
+			return newAlias
+		}
+	}
+	// Should never reach here
+}
+
+// UsingGoImports should be used with basic packages like "time", or "context"
+func (p *OrmPlugin) UsingGoImports(pkgNames ...string) {
+	p.GetFileImports().stdImports = append(p.GetFileImports().stdImports, pkgNames...)
+}
+
+// UsingImports should only be used with the specially defined import vars
+func (p *OrmPlugin) UsingImports(pkgs ...*pkgImport) {
+	for _, pkg := range pkgs {
+		p.GetFileImports().packages[pkg.alias] = pkg
+	}
+}
+
 type fileImports struct {
 	wktPkgName      string
-	usingGORM       bool
-	usingUUID       bool
-	usingTime       bool
-	usingAuth       bool
-	usingJSON       bool
 	typesToRegister []string
+	stdImports      []string
+	packages        map[string]*pkgImport
+}
+
+func newFileImports() *fileImports {
+	return &fileImports{packages: make(map[string]*pkgImport)}
+}
+
+func (p *OrmPlugin) GetFileImports() *fileImports {
+	return p.fileImports[p.currentFile]
 }
 
 // GenerateImports writes out required imports for the generated files
 func (p *OrmPlugin) GenerateImports(file *generator.FileDescriptor) {
 	imports := p.fileImports[file]
-	stdImports := []string{}
-	githubImports := map[string]string{}
-	if imports.usingGORM {
-		stdImports = append(stdImports, "context", "errors")
-		githubImports["gorm"] = "github.com/jinzhu/gorm"
-		githubImports["tkgorm"] = "github.com/infobloxopen/atlas-app-toolkit/gorm"
-	}
-	if imports.usingUUID {
-		githubImports["uuid"] = "github.com/satori/go.uuid"
-		githubImports["gtypes"] = "github.com/infobloxopen/protoc-gen-gorm/types"
-	}
-	if imports.usingTime {
-		stdImports = append(stdImports, "time")
-		githubImports["ptypes"] = "github.com/golang/protobuf/ptypes"
-	}
-	if imports.usingAuth {
-		githubImports["auth"] = "github.com/infobloxopen/atlas-app-toolkit/auth"
-	}
-	if imports.usingJSON {
-		if p.dbEngine == ENGINE_POSTGRES {
-			githubImports["gormpq"] = "github.com/jinzhu/gorm/dialects/postgres"
-			githubImports["gtypes"] = "github.com/infobloxopen/protoc-gen-gorm/types"
-		}
-	}
 	for _, typeName := range imports.typesToRegister {
 		p.RecordTypeUse(typeName)
 	}
-	sort.Strings(stdImports)
-	for _, dep := range stdImports {
+	githubImports := imports.packages
+	sort.Strings(imports.stdImports)
```

Comment:
`Gofmt` does sorting of imports which is called for each generated file. Is it not enough for us?

Reply:
I haven't noticed that the imports are automatically sorted. 
I just removed the sort statements and rebuilt and got this:
```
import time "time"
import context "context"
import errors "errors"

import gorm2 "github.com/infobloxopen/atlas-app-toolkit/gorm"
import ptypes1 "github.com/golang/protobuf/ptypes"
import auth1 "github.com/infobloxopen/atlas-app-toolkit/auth"
import gorm1 "github.com/jinzhu/gorm"
```

Reply:
oops, I'm sorry for misinformation.